generate/generate.go

```diff
@@ -501,6 +510,267 @@ func (g *Generator) fileOptions(pkg *loader.GunkPackage) (*desc.FileOptions, err
+					if !ok {
+						return nil, fmt.Errorf("cannot found security scheme in: %s", n.Value.(*ast.SelectorExpr).Sel.Name)
+					}
+					value.In = options.SecurityScheme_In(sel)
+				case "Flow":
+					sel, ok := options.SecurityScheme_Flow_value[n.Value.(*ast.SelectorExpr).Sel.Name]
+					if !ok {
+						return nil, fmt.Errorf("cannot found security scheme flow: %s", n.Value.(*ast.SelectorExpr).Sel.Name)
+					}
+					value.Flow = options.SecurityScheme_Flow(sel)
+				case "Scopes":
+					scope := map[string]string{}
+					for _, sc := range n.Value.(*ast.CompositeLit).Elts {
+						for _, e := range sc.(*ast.KeyValueExpr).Value.(*ast.CompositeLit).Elts {
+							kv := e.(*ast.KeyValueExpr)
+							strKey, _ := strconv.Unquote(kv.Key.(*ast.BasicLit).Value)
+							strValue, _ := strconv.Unquote(kv.Value.(*ast.BasicLit).Value)
+							scope[strKey] = strValue
+						}
+					}
+					value.Scopes = &options.Scopes{
+						Scope: scope,
+					}
+				}
+			}
+			sd.Security[key] = value
+		}
+	}
+	return sd, nil
+}
+
+func (g *Generator) convertExternalDocs(lit *ast.CompositeLit) (*options.ExternalDocumentation, error) {
+	externalDocs := &options.ExternalDocumentation{}
+	for _, valElt := range lit.Elts {
+		kvExpr := valElt.(*ast.KeyValueExpr)
+		key := kvExpr.Key.(*ast.Ident).Name
+		val, _ := strconv.Unquote(kvExpr.Value.(*ast.BasicLit).Value)
+		if err := utils.SetValue(key, externalDocs, val); err != nil {
+			return nil, err
+		}
+	}
+	return externalDocs, nil
+}
+
+func (g *Generator) convertInfo(lit *ast.CompositeLit) (*options.Info, error) {
+	info := &options.Info{}
+	for _, valElt := range lit.Elts {
+		kvExpr := valElt.(*ast.KeyValueExpr)
+		key := kvExpr.Key.(*ast.Ident).Name
+		switch key {
+		case "Title",
+			"Description",
+			"Version":
+			val, _ := strconv.Unquote(kvExpr.Value.(*ast.BasicLit).Value)
+			if err := utils.SetValue(key, info, val); err != nil {
+				return nil, err
+			}
+		case "Contact":
+			contact := &options.Contact{}
+			val := kvExpr.Value.(*ast.CompositeLit)
+			for _, cElt := range val.Elts {
+				kvExpr := cElt.(*ast.KeyValueExpr)
+				val, _ := strconv.Unquote(kvExpr.Value.(*ast.BasicLit).Value)
+				if err := utils.SetValue(kvExpr.Key.(*ast.Ident).Name, contact, val); err != nil {
+					return nil, err
+				}
+			}
+			info.Contact = contact
+		case "License":
+			license := &options.License{}
+			val := kvExpr.Value.(*ast.CompositeLit)
+			for _, lElt := range val.Elts {
+				kvExpr := lElt.(*ast.KeyValueExpr)
+				val, _ := strconv.Unquote(kvExpr.Value.(*ast.BasicLit).Value)
+				if err := utils.SetValue(kvExpr.Key.(*ast.Ident).Name, license, val); err != nil {
+					return nil, err
+				}
+			}
+			info.License = license
+		}
+	}
+	return info, nil
+}
+
+func (g *Generator) convertJSONSchema(lit *ast.CompositeLit) (*options.JSONSchema, error) {
+	jsonSchema := &options.JSONSchema{}
+	for _, elt := range lit.Elts {
+		name := elt.(*ast.KeyValueExpr).Key.(*ast.Ident).Name
```

Comment:
you can deduplicate all this `elt.(*ast.KeyValueExpr)` work, here and elsewhere, like:

```
name := elt.(*ast.KeyValueExpr).Key.(*ast.Ident).Name
value := elt.(*ast.KeyValueExpr).Value
```